swagger/public_v1/template.yml

```diff
@@ -20,46 +20,106 @@ components:
     Sort:
       type: array
       example: "provider.provider_name,name"
+      description: "This schema is used to sort collections."
       items:
         type: string
         example: "name"
+        description: "A comma delimited string of fields to sort the collection by. For example `provider.provider_name,name`."
     Filter:
       type: object
       example: ""
+      description: "This schema is used to search within collections to return more specific results."
       properties:
         has_vacancies:
           description: "Return courses that only have vacancies?"
-          type: string
-          example: "true"
+          type: boolean
+          example: true
         funding:
           description: "Return courses that are salary funded."
           type: string
           example: "salary"
+          enum:
+            - salary
+            - apprenticeship
+            - fee
         qualification:
           description: "Search courses based on the award given on course completion."
           type: array
           example: "qts,pgce,pgde"
+          enum:
+            - qts
+            - pgce_with_qts
+            - pgde
+            - pgce
+            - pgde_with_qts
           items:
             type: string
             example: "qts"
         study_type:
           description: "Search full time or part time courses or both."
           type: array
           example: "full_time,part_time"
+          enum:
+            - full_time
+            - part_time
+            - full_time_or_part_time
           items:
             type: string
             example: "full_time"
         subjects:
           description: "Returns courses that include at least one of the given subjects."
           type: array
           example: "00,01,W1"
+          enum:
+            - 00
+            - 01
+            - 02
+            - 03
+            - 04
+            - 06
+            - 07
+            - W1
+            - F0
+            - C1
+            - 08
+            - F1
+            - 09
+            - Q8
+            - P3
+            - 11
+            - 12
+            - DT
+            - 13
+            - L1
+            - Q3
+            - F8
+            - L5
+            - V1
+            - G1
+            - W3
+            - P1
+            - C6
+            - F3
+            - C8
+            - V6
+            - 14
+            - 15
+            - 16
+            - 17
+            - 18
+            - 19
+            - 20
+            - 21
+            - 22
+            - 41
+            - 24
```

Comment:
these needs to be quote wrapped....

json generated has mixed types

Reply:
good spot